modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_9_1/DDMStructureUpgradeProcess.java

```diff
@@ -122,30 +129,35 @@ private void _upgradeDDMStructure() throws Exception {
 	private void _upgradeDDMStructureVersion() throws Exception {
 		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
 				StringBundler.concat(
-					"select DDMStructureVersion.structureVersionId, ",
+					"select DDMStructureVersion.ctCollectionId, ",
+					"DDMStructureVersion.structureVersionId, ",
 					"DDMStructureVersion.definition from DDMStructure inner ",
-					"join DDMStructureVersion on DDMStructure.structureId = ",
+					"join DDMStructureVersion on DDMStructure.ctCollectionId ",
+					"= DDMStructureVersion.ctCollectionId and ",
+					"DDMStructure.structureId = ",
 					"DDMStructureVersion.structureId where ",
 					"DDMStructure.classNameId = ?"));
 			PreparedStatement preparedStatement2 =
 				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
 					connection,
 					"update DDMStructureVersion set definition = ? where " +
-						"structureVersionId = ?")) {
+						"ctCollectionId = ? and structureVersionId = ?")) {
 
 			preparedStatement1.setLong(
 				1, PortalUtil.getClassNameId(DDMFormInstance.class.getName()));
 
 			try (ResultSet resultSet = preparedStatement1.executeQuery()) {
 				while (resultSet.next()) {
+					long ctCollectionId = resultSet.getLong("ctCollectionId");
 					long structureVersionId = resultSet.getLong(
 						"structureVersionId");
 
 					preparedStatement2.setString(
 						1,
 						_upgradeDDMStructureVersionDefinition(
 							resultSet.getString("definition")));
-					preparedStatement2.setLong(2, structureVersionId);
+					preparedStatement2.setLong(2, ctCollectionId);
+					preparedStatement2.setLong(3, structureVersionId);
```

Comment:
Can you inline ctCollectionId and structureVersionId?

Reply:
Yep